### PR TITLE
[XPU] support alltoall_single

### PR DESCRIPTION
**Pull request by XiaociZhang**
### PR Category

Communication Library

### PR Types

New features

### Description

XPU support alltoall_single.
alltoallv not covered since it's not supported by XCCL.

### CODE REVIEW

File: paddle/phi/backends/xpu/xpu3_op_list.cc [hunk 1]
```diff
@@ -111,6 +111,14 @@ XPUOpMap& get_kl3_ops() {
...
                      phi::DataType::UINT8,
                      phi::DataType::BOOL})},
+      {"c_alltoall",
+       XPUKernelSet({phi::DataType::FLOAT16,
+                     phi::DataType::FLOAT32,
+                     phi::DataType::FLOAT64,
+                     phi::DataType::INT32,
+                     phi::DataType::INT64,
+                     phi::DataType::UINT8,
+                     phi::DataType::BOOL})},
```
Comment (lj970926) [inline paddle/phi/backends/xpu/xpu3_op_list.cc:121]: BF16是BKCL还不支持吗？

**Review by lj970926 — APPROVED** [review]
LGTM

File: paddle/phi/backends/xpu/xpu3_op_list.cc (hunk 1 above)
Comment (HarperCy) [reply paddle/phi/backends/xpu/xpu3_op_list.cc:121]: 我看单测里按照uint16传入的

**Review by HarperCy — APPROVED** [review]

File: paddle/phi/backends/xpu/xpu3_op_list.cc (hunk 1 above)
Comment (XiaociZhang) [reply paddle/phi/backends/xpu/xpu3_op_list.cc:121]: 应该所有接口都支持bf16了，后面再提个PR统一加下。

**Review by houj04 — APPROVED** [review]

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**